### PR TITLE
feat(ui5-toast): migrate to Typescript

### PR DESCRIPTION
**Pull request by elenastoyanovaa**

### CODE REVIEW

**Review by ilhan007 — APPROVED** [review]
Looks good to me.

**Comment by ilhan007** [comment]
Maybe  change it to feat, so that it becomes part of the release notes

File: packages/main/src/Toast.ts [hunk 1]
```diff
@@ -142,10 +53,84 @@ const metadata = {
...
+	 * <br><br>
+	 * <b>Note:</b> Although this slot accepts HTML Elements, it is strongly recommended that you only use text in order to preserve the intended design.
+	 *
+	 * @type {Node[]}
+	 * @slot
+	 * @public
+	 * @name sap.ui.webc.main.Toast.prototype.default
+	 */
+
+	_reopen!: boolean;
```
[the remark this answers is not in the document]
Comment (ilhan007) [reply packages/main/src/Toast.ts:133]: nice catch `_reopen: boolean;` is enough

**Comment by ilhan007** [comment]
This seems to be the next unstable test:

Input.spec.js
1) Testing events Should fire input event with correctly swiped values
The input event is fired with the correct values: expected '3' to equal '2'

**Comment by ndeshev** [comment]
> This seems to be the next unstable test:
> 
> Input.spec.js
> 
> 1. Testing events Should fire input event with correctly swiped values
>    The input event is fired with the correct values: expected '3' to equal '2'

I noticed that too, the failing test is from the RangeSlider testsuite, I created a pull-request temporary to temporarily disable it - #6508

**Comment by ilhan007** [comment]
Related to: https://github.com/SAP/ui5-webcomponents/issues/4337

**Pull request merged**